commands/utils/params.go

```diff
@@ -2,174 +2,160 @@ package utils
 
 import (
 	"fmt"
+	"github.com/jfrog/build-info-go/utils"
+	"github.com/jfrog/froggit-go/vcsclient"
+	"github.com/jfrog/froggit-go/vcsutils"
+	coreconfig "github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"gopkg.in/yaml.v3"
 	"os"
+	"path/filepath"
 	"strconv"
 	"strings"
+)
 
-	"github.com/jfrog/froggit-go/vcsclient"
-	"github.com/jfrog/froggit-go/vcsutils"
-	coreconfig "github.com/jfrog/jfrog-cli-core/v2/utils/config"
+const (
+	FrogbotConfigFile = "frogbot-config.yml"
 )
 
-type FrogbotParams struct {
-	JFrogEnvParams
-	GitParams
-	ScanPullRequestParams
-	WorkingDirectory   string
-	InstallCommandName string
-	InstallCommandArgs []string
-	RequirementsFile   string
-}
+var configRelativePath = filepath.Join(".", ".jfrog", FrogbotConfigFile)
 
-type ScanPullRequestParams struct {
-	IncludeAllVulnerabilities bool
+type FrogbotConfigAggregator []FrogbotRepoConfig
+
+type FrogbotRepoConfig struct {
+	Server                    coreconfig.ServerDetails
 	SimplifiedOutput          bool
-	FailOnSecurityIssues      bool
-	UseWrapper                bool
+	IncludeAllVulnerabilities bool      `yaml:"includeAllVulnerabilities,omitempty"`
+	FailOnSecurityIssues      bool      `yaml:"failOnSecurityIssues,omitempty"`
+	JFrogProjectKey           string    `yaml:"jfrogProjectKey,omitempty"`
+	Projects                  []Project `yaml:"projects,omitempty"`
+	Watches                   []string  `yaml:"watches,omitempty"`
+	GitParams                 `yaml:"gitParams,omitempty"`
 }
 
-type JFrogEnvParams struct {
-	Server  coreconfig.ServerDetails
-	Project string
-	Watches string
+type Project struct {
+	InstallCommandName  string   `yaml:"installCommandName,omitempty"`
+	InstallCommandArgs  []string `yaml:"installCommandArgs,omitempty"`
+	PipRequirementsFile string   `yaml:"pipRequirementsFile,omitempty"`
+	WorkingDirs         []string `yaml:"workingDirs,omitempty"`
+	UseWrapper          bool     `yaml:"useWrapper,omitempty"`
 }
 
 type GitParams struct {
```

Comment:
What do you think about separating the GitParams (RepoName, Branches), JFrogParams (JFrogProjectKey, Watches) and FrogbotParams in the `frogbot-config.yml` schema? I mean something like this:
```yaml
# JFrog Frogbot Config file supports multiple repository configuration.
# It is important to note that 'scan-pull-request' and 'create-fix-pull-request' only work on a single repository, thus only the first repository in the configuration file will be used.
# Please use 'scan-pull-requests' command to scan the multiple repository pull requests configurations.

- jfrogParams:
    - jfrogProjectKey: "" # JFrog project key. Learn more about it here: https://www.jfrog.com/confluence/display/JFROG/Projects. Optional.
      watches: # Xray Watches. Learn more about them here: https://www.jfrog.com/confluence/display/JFROG/Configuring+Xray+Watches. Optional.
        - ""
  gitParams: # Git parameters to base the scan on
    repoName: "" # Name of the repository to scan. Mandatory.
    branches: # List of branches to scan. Mandatory for scan-and-fix-repos/create-fix-pull-requests commands.
      - master
  frogbotParams:
    - projects: # List of package manager related projects (e.g. npm, pip, maven projects) with their relevant working directories, installation command, etc.
      - installCommandName: "" # Installation command (e.g. npm, maven, yarn). Mandatory only for projects which use npm, yarn 2, nuget and dotnet to download their dependencies.
        installCommandArgs: # List of arguments to the installation command (e.g. install, i)
          - ""
        workingDirs: # List of relative path to the projects directories in the git repository. [Default: root directory] Optional.
          - "."
        pipRequirementsFile: "" # The requirements file name that used to install dependencies in case of pip package manager. (pip install -r <JF_REQUIREMENTS_FILE>) [Default: pip install .] Optional.
        useWrapper: true # Use Gradle Wrapper (gradlew/gradlew.bat) to run Gradle. [Default: true] Optional.
      includeAllVulnerabilities: false # Displays all existing vulnerabilities, including the ones that were added by the pull request. [Default: false] Optional.
      failOnSecurityIssues: false # Fails the Frogbot task if any security issue is found. [Default: false] Optional.
```

Even more clearly, I think.